File: sdk/chocobase-kt/src/main/kotlin/io/chocobase/ChocoClient.kt
package io.chocobase

import okhttp3.OkHttpClient

class ChocoClient(
    val url: String,
    val apiKey: String,
    customHeaders: Map<String, String>? = null
) {
    val httpClient = OkHttpClient()
    val headers = mutableMapOf(
        "apikey" to apiKey,
        "Authorization" to "Bearer $apiKey",
        "Content-Type" to "application/json"
    ).apply {
        customHeaders?.let { putAll(it) }
    }

    val auth = AuthClient(url, httpClient, headers)
    val storage = StorageClient(url, httpClient, headers)
    val functions = FunctionsClient(url, httpClient, headers)

    fun from(table: String): QueryBuilder {
        return QueryBuilder(url, table, httpClient, headers)
    }
}

fun createClient(url: String, apiKey: String, customHeaders: Map<String, String>? = null): ChocoClient {
    return ChocoClient(url, apiKey, customHeaders)
}

File: sdk/chocobase-php/src/ChocoClient.php
<?php

namespace ChocoBase;

class ChocoClient
{
    private string $url;
    private string $apiKey;
    private array $headers;

    public Auth $auth;
    public Storage $storage;
    public Functions $functions;

    public function __construct(string $url, string $apiKey, array $customHeaders = [])
    {
        $this->url = rtrim($url, '/');
        $this->apiKey = $apiKey;
        $this->headers = array_merge([
            'apikey' => $apiKey,
            'Authorization' => "Bearer {$apiKey}",
            'Content-Type' => 'application/json'
        ], $customHeaders);

        $this->auth = new Auth($this->url, $this->headers);
        $this->storage = new Storage($this->url, $this->headers);
        $this->functions = new Functions($this->url, $this->headers);
    }

    public function from(string $table): Postgrest
    {
        return new Postgrest($this->url, $table, $this->headers);
    }

    public static function createClient(string $url, string $apiKey, array $customHeaders = []): self
    {
        return new self($url, $apiKey, $customHeaders);
    }
}
